deploy/kubernetes/filebeat-kubernetes.yaml

```diff
@@ -113,14 +113,15 @@ data:
   filebeat.yml: |-
     filebeat.inputs:
     - type: filestream
-      id: kubernetes-container-logs-${data.kubernetes.pod.name}-${data.kubernetes.container.id}
+      id: kubernetes-container-logs
```

Comment:
These substitutions should work only in autodiscover, not in regular input configuration, AFAIK.

I see @gsantoro made this change in https://github.com/elastic/beats/pull/37401
How was it tested?

Reply:
@constanca-m My point is rather if this was not tested, how can we trust the rest of the changes from that PR? Perhaps we need to ask @gsantoro to re-test those changes.

Reply:
hello @rdner , sorry that was my fault. It was an unintended change. I'll do some testing now anyway

Reply:
I have now tested this. it works as expected with this PR changes. thanks @constanca-m for catching this